packages/dashboard/src/components/widgets/widget.tsx

```diff
@@ -29,8 +39,31 @@ const WidgetComponent: React.FC<WidgetProps> = ({
 }) => {
   const { x, y, z, width, height } = widget;
 
+  // TODO: Replace with Redux dispatch
+  const [assets, setAssets] = useState<null | AssetsParam>(null);
+  const [internalWidget, setInternalWidget] = useState(widget);
+
+  const [, drop] = useDrop(
+    () => ({
+      accept: ItemTypes.ResourceExplorerAssetProperty,
+      drop: ({ queryAssetsParam }: { queryAssetsParam: AssetsParam }) => {
+        setAssets(queryAssetsParam);
```

Comment:
The update widget action should be available, is it possible to dispatch that from here appending the queryAssetsParam to assets?

Reply:
I am planning on doing this as a followup, originally with a separate branch I wrote but now potentially integrating it with Bill's branch that does this.

Reply:
The reason I want to separate it out is that it's a little more complicated -- we want to be able to drag multiple assets and multiple asset properties onto a widget, so it has to both be able to append assets as well as determine if asset properties being passed belong to assets already attached to the widget, and if so, append the property to the existing asset rather than appending a duplicate asset.

Reply:
Ahhh ok that makes sense. Plus with dan's work on kpis, sometimes we want to replace the asset property instead of append it.